components/st-microelectronics/stm-api.stanza

```diff
@@ -130,132 +138,177 @@ defn match-pin? (pin-ref:Ref power:String) :
   if p-ref == power :
     true
 
-defn find-pins-list (obj:JITXObject my-pin:String) :
+; return list of pins that match name (exactly) / don't check for empty
+defn get-pin-list (obj:JITXObject pin-id:?) :
+  val pin-name = to-string(pin-id)
   val result = to-tuple $ for p in pins(obj) filter :
-    if substring?(to-string(ref(p)), my-pin) :
-      match-pin?(ref(p), my-pin)
+    if substring?(to-string(ref(p)), pin-name) :
+      match-pin?(ref(p), pin-name)
   result
 
-defn find-pins? (obj:JITXObject my-pin:String) :
-  val result = to-tuple $ for p in pins(obj) filter :
-    if substring?(to-string(ref(p)), my-pin) :
-      match-pin?(ref(p), my-pin)
+; find the pin exact match return a boolean
+defn find-pin? (obj:JITXObject my-pin:String|PwrPinNames|GndPinNames|ResetPinNames|BootPinNames|ReferencePinNames) :
```

Comment:
I recommend creating a new type that each PinNames subtypes.
```
public deftype STMPinNames
public pcb-enum .../PwrPinNames <: STMPinNames
...
```

```
defn get-pin-list (obj:JITXObject pin-id:String|STMPinNames)
defn find-pin? (obj:JITXObject my-pin:String|STMPinNames)
```

Reply:
This is a great suggestion for readability. Done.